hknweb/events/views.py

```diff
@@ -41,27 +62,34 @@ def rsvp(request, id):
     rsvps = event.rsvp_set.count()
 
     if request.user.is_authenticated and (event.rsvp_limit is None or rsvps < event.rsvp_limit):
-        Rsvp.objects.create(user=request.user, event=event)
+        Rsvp.objects.create(user=request.user, event=event, confirmed=False)
         messages.success(request, 'RSVP\'d!')
     else:
         messages.error(request, 'Could not RSVP; the RSVP limit has been reached.')
     return redirect('/events/' + str(id))
 
 @csrf_exempt  #doing this for now bc idk how to make csrf work
+@login_required(login_url='/accounts/login/')
+@check_account_access
 def unrsvp(request, id):
     if request.method != 'POST':
         raise Http404()
 
     event = get_object_or_404(Event, pk=id)
 
     if request.user.is_authenticated:
-        rsvp = get_object_or_404(Rsvp, user=request.user, event=event)
-        rsvp.delete()
-        messages.success(request, 'un-RSVP\'d :(')
+        # check if rsvp for this event and this user already exists; if false, then set true
+        if datetime.datetime.now().replace(tzinfo=pytz.UTC) > event.end_time.replace(tzinfo=pytz.UTC):
```

Comment:
https://docs.djangoproject.com/en/2.2/topics/i18n/timezones/

The better timezone-aware strategy would be to use the Django timezone import:
```
from django.utils import timezone
now = timezone.now()
```

Reply:
LOL that ridiculously outdated comment is still here